src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -143,8 +143,7 @@ private static boolean isElse(DetailAST ast) {
      * @return whether the toke does represent an SLIST as part of an ELSE
      */
     private static boolean isElseWithCurlyBraces(DetailAST ast) {
```

Comment:
This method is used in `NestedIfDepthCheck` only which makes sense to move it there as for me.

Regarding `NestedIfDepthCheck`, it is super weird check:
```
// nesting == 2
if (true) {
} else {
    if (true) {
    } else {
        if (false) { // I would expect violation here, but I get nothing
        } else {
        }
    }
}

// nesting == 2
if (true) {
    if (true) {
        if (false) { // violation here occurred as expected
        } else {
        }
    } else {
    }
} else {
}
```
@romani do you know why this check handles only nested `if` and ignores nesting in `else` blocks?

Reply:
I do not see any good reason https://checkstyle.org/checks/coding/nestedifdepth.html#NestedIfDepth , it is  defect.
@rdiachenko , please create issue.

------

if util method is used only in one Check, it should be moved to Check and covered.

Reply:
Done The method is moved to check itself

Reply:
@Kevin222004 , please create issue to speed up a process and share link here

Reply:
Issue created: https://github.com/checkstyle/checkstyle/issues/13493